src/meta/src/hummock/manager/mod.rs

```diff
@@ -878,259 +878,338 @@ impl HummockManager {
+                    task_type: compact_task.compaction_task_type as i32,
+                    split_weight_by_vnode: vnode_partition_count,
+                    ..Default::default()
+                };
+
+                let is_trivial_reclaim = CompactStatus::is_trivial_reclaim(&compact_task);
+                let is_trivial_move = CompactStatus::is_trivial_move_task(&compact_task);
+                if is_trivial_reclaim || (is_trivial_move && can_trivial_move) {
+                    let log_label = if is_trivial_reclaim {
+                        "TrivialReclaim"
+                    } else {
+                        "TrivialMove"
+                    };
+                    let label = if is_trivial_reclaim {
+                        "trivial-space-reclaim"
+                    } else {
+                        "trivial-move"
+                    };
+
+                    tracing::debug!(
+                        "{} for compaction group {}: input: {:?}, cost time: {:?}",
+                        log_label,
+                        compact_task.compaction_group_id,
+                        compact_task.input_ssts,
+                        start_time.elapsed()
+                    );
+                    compact_task.set_task_status(TaskStatus::Success);
+                    compact_status.report_compact_task(&compact_task);
+                    if !is_trivial_reclaim {
+                        compact_task.sorted_output_ssts =
+                            compact_task.input_ssts[0].table_infos.clone();
+                    }
+                    self.metrics
+                        .compact_frequency
+                        .with_label_values(&[
+                            label,
+                            &compact_task.compaction_group_id.to_string(),
+                            selector.task_type().as_str_name(),
+                            "SUCCESS",
+                        ])
+                        .inc();
+                    let version_delta = gen_version_delta(
+                        &mut hummock_version_deltas,
+                        &mut branched_ssts,
+                        &current_version,
+                        &compact_task,
+                        deterministic_mode,
+                    );
+                    current_version.apply_version_delta(&version_delta);
+                    trivial_tasks.push(compact_task);
+                    if trivial_tasks.len() >= MAX_TRIVIAL_MOVE_TASK_COUNT {
+                        break 'outside;
+                    }
+                } else {
+                    if group_config.compaction_config.split_weight_by_vnode > 0 {
+                        for table_id in &compact_task.existing_table_ids {
+                            compact_task
+                                .table_vnode_partition
+                                .insert(*table_id, vnode_partition_count);
+                        }
```

Comment:
This is a question instead of a comment because this logic existed prior to this PR: is it possible to enter this branch for cg2 and cg3 (cg with more than one `existing_table_ids`)? If yes, using `compact_task.input.vnode_partition_count` for all tables looks strange to me.

Reply:
No, because `split_weight_by_vnode` of cg2 and cg2 will always equals to zero.